backend/cloud/app.js

```diff
@@ -63,6 +65,25 @@ app.get("/t/:slug", async function inner(req, res, next) {
 app.get("/u/:id", makeWithOG(User, []));
 app.get("/a/:id", makeWithOG(Activity, ["objects", "team"]));
 
+app.get("/app-redir", (req, res) => {
+  const hostname = req.hostname
+  const pageConfig = config.pages[hostname]

```

Comment:
fallback for when `hostname` doesn't resolve to a config:
```suggestion
  const pageConfig = config.pages[hostname];
  if (!pageConfig) {
    res.redirect("/");
    return
  }

```